server/player/player.go

```diff
@@ -1644,9 +1644,15 @@ func (p *Player) UseItemOnEntity(e world.Entity) bool {
 // have.
 // If the player cannot reach the entity at its position, the method returns immediately.
 func (p *Player) AttackEntity(e world.Entity) bool {
-	if !p.canReach(e.Position()) {
+	if !p.canReach(e.Position()) || p.Dead() {
+		return false
+	}
+
+	living, ok := e.(entity.Living)
+	if !ok || living.Dead() {
 		return false
 	}
```

Comment:
Moving the living check here means the `HandleAttackEntity` event will not be called. Can you not add the dead check to where the if statement was originally? Or you can keep both checks & change this one here to be `ok && living.Dead()`

Reply:
changed to `ok && living.Dead()`